src/librustdoc/doctest.rs

```diff
@@ -119,7 +119,7 @@ crate fn run(options: Options) -> Result<(), ErrorReported> {
                 let crate_attrs = tcx.hir().attrs(CRATE_HIR_ID);
 
                 let mut opts = scrape_test_config(crate_attrs);
-                opts.display_warnings |= options.display_warnings;
+                opts.display_doctest_warnings |= options.display_doctest_warnings;
```

Comment:
Wait, do we allow setting this both through a CLI flag and a crate attribute? That seems weird, why not just one or the other?

Reply:
A bit weird to have this through a crate attribute but I guess it's convenient to enable it automatically in your `Cargo.toml` file with a feature under a profile.

Reply:
Ok. I think we should remove this as an attribute and make it a CLI flag only, I don't think there's much benefit in making it a feature since you'd need to pass `--feature doctest-warnings` or whatever anyway.

Reply:
Should I do it in this PR or open an issue?

Reply:
Umm. This PR has probably been in limbo long enough, an issue is fine.

Reply:
I opened https://github.com/rust-lang/rust/issues/88792.